tsl/src/remote/connection.c

```diff
@@ -1384,37 +1593,95 @@ remote_connection_open_with_options_nothrow(const char *node_name, List *connect
 	if (NULL == pg_conn)
 		return NULL;
 
-	if (PQstatus(pg_conn) != CONNECTION_OK)
+	if (PQstatus(pg_conn) == CONNECTION_BAD)
 	{
 		finish_connection(pg_conn, errmsg);
 		return NULL;
 	}
 
-	ts_conn = remote_connection_create(pg_conn, false, node_name);
+	status = PGRES_POLLING_WRITING;
+
+	do
```

Comment:
I guess you could wrap all of below by calling `wait_for_consume_input()` on a read event

Reply:
No, connection establishment needs special handling (nothing to flush or consume). Also see the comment below about not reusing a socket, since the socket might be changed in each iteration of the loop.